src/renderer/api/catalog-entity-registry.ts

```diff
@@ -27,15 +27,25 @@ import type { Cluster } from "../../main/cluster";
 import { ClusterStore } from "../../common/cluster-store";
 import { Disposer, iter } from "../utils";
 import { once } from "lodash";
+import type { CatalogEntityItem } from "../../renderer/components/+catalog/catalog-entity-item";
 
 export type EntityFilter = (entity: CatalogEntity) => any;
+export type CatelogEntityOnRunHook = (entity: CatalogEntity | CatalogEntityItem<CatalogEntity>) => boolean | Promise<boolean>;
```

Comment:
spelling mistake: should be catalog...

Reply:
Also I think that leaking `CatalogEnityItem` to the extension API is a very bad idea.

Reply:
Thanks, fix the typo. About `CatalogEnityItem`, it more about reflecting the true types... but I guess we can remove it.